### PR TITLE
DeviceInfo: Use config file for device specific settings.

### PR DESCRIPTION
**Pull request by MagneFire**
Move away from using compile time watch specifics. Use a configuration file `machine.conf` if it is provided by a `meta-*-hybris` layer.

Main benefits:
- Won't have to compile `qml-asteroid` for every watch.
- Allows for configuration changes that aren't recommended (i.e. disable burn-in-protection on OLED displays).

I have created a `machine.conf` for `sturgeon` (https://github.com/AsteroidOS/meta-sturgeon-hybris/pull/13). Depending on changes required I will also adjust the other `hybris` layers when no more changes are required.

### CODE REVIEW

File: src/utils/src/deviceinfo.cpp [hunk 1]
```diff
@@ -16,49 +16,49 @@
...
+#include <QFile>
+#include <QCoreApplication>
+
+const char* CONFIG_FILE = "/etc/machine.conf";
+
+DeviceInfo::DeviceInfo()
+{
+    QCoreApplication::setOrganizationName("AsteroidOS");
+    QCoreApplication::setOrganizationDomain("asteroidos.org");
+    QCoreApplication::setApplicationName("Device Info");
```
Comment (FlorentRevest) [inline src/utils/src/deviceinfo.cpp:28]: Mhhh why is this necessary ? Aren't those set in here already https://github.com/AsteroidOS/qml-asteroid/blob/master/src/app/asteroidapp.cpp#L65 ?
DeviceInfo should not set the application name of a global singleton to "Device Info".

Comment (FlorentRevest) [reply src/utils/src/deviceinfo.cpp:28]: If lipstick doesn't set those, then those should be set in Lipstick, not in a QML Item

File: src/utils/src/deviceinfo.cpp [hunk 2]
```diff
@@ -16,49 +16,49 @@
  */
 
 #include "deviceinfo.h"
+#include <QFile>
+#include <QCoreApplication>
+
+const char* CONFIG_FILE = "/etc/machine.conf";
```
Comment (FlorentRevest) [inline src/utils/src/deviceinfo.cpp:22]: Let's make a subdirectory for qml-asteroid. Or at least for asteroid. I don't like using the /etc namespace for something as generic as "machine.conf"

File: src/utils/src/deviceinfo.cpp [hunk 3]
```diff
@@ -16,49 +16,49 @@
...
+
+DeviceInfo::DeviceInfo()
+{
+    QCoreApplication::setOrganizationName("AsteroidOS");
+    QCoreApplication::setOrganizationDomain("asteroidos.org");
+    QCoreApplication::setApplicationName("Device Info");
+    m_settings = new QSettings(CONFIG_FILE, QSettings::IniFormat);
```
Comment (FlorentRevest) [inline src/utils/src/deviceinfo.cpp:29]: If you want a heap-allocated QObject to get garbage collected automatically you should provide a parent. Otherwise this might cause a memory leak. I suspect we have tons of those in the codebase already because I wasn't particularly careful with those in the past. Maybe you can also just have m_settings be a simple QSettings member.

**Review by FlorentRevest — CHANGES_REQUESTED** [review]

File: src/utils/src/deviceinfo.cpp (hunk 1 above)
Comment (MagneFire) [reply src/utils/src/deviceinfo.cpp]: You are correct, this wasn't necessary.

File: src/utils/src/deviceinfo.cpp (hunk 2 above)
Comment (MagneFire) [reply src/utils/src/deviceinfo.cpp]: Moved to a subdirectory named `qml-asteroid`.

File: src/utils/src/deviceinfo.cpp (hunk 3 above)
Comment (MagneFire) [reply src/utils/src/deviceinfo.cpp]: Now a simple member.

**Review by FlorentRevest — APPROVED** [review]

**Comment by FlorentRevest** [comment]
Approved but pending your answer on https://github.com/AsteroidOS/meta-asteroid/pull/50/files#r502890018
